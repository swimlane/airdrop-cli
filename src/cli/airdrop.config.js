module.exports = {
  defaults: {
    package_path: './-/',
    package_root: '/-/',    
  },
  name: 'airdrop'
}