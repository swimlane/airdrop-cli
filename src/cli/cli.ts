import { build, GluegunToolbox } from 'gluegun';

/**
 * Create the cli and kick it off
 */
export async function run(argv?: string[] | string): Promise<GluegunToolbox> {
  // create a CLI runtime
  const cli = build('airdrop')
    .src(__dirname)
    .help()
    .version()
    .exclude([
      'semver',
      'system', 
      'prompt',
      'http',
      'template',
      'patching'
    ])
    .defaultCommand({
      run: async ({ print: { info }, meta: { version }, runtime: { brand } }: GluegunToolbox) => {
        info(`${brand} version ${version()}`)
        info(``)
        info(`  Type ${brand} --help for more info`)
      }
    })
    .command({
      name: 'add-bundle',
      alias: ['ab'],
      run: async (toolbox: GluegunToolbox) => {
        const { parameters, print } = toolbox;
        const force = parameters.options.force ? '--force' : '';
        const optimize = parameters.options.optimize ? '--optimize' : '';

        print.info('Adding packages');
        await cli.run(`a ${parameters.string} ${force}`);

        print.info('Bundling packages');
        await cli.run(`b ${parameters.string} ${force} ${optimize}`);
      }
    })
    .create();

  return cli.run(argv);
}
