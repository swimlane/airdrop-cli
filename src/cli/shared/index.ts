export * from './getMap';
export * from './addPackages';
export * from './bundlePackages';