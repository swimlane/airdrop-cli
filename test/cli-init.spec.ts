import { createSandbox, execPkgdrop } from './cli.util';

describe('init', () => {
  let output: string;
  let sandbox: any;

  beforeAll(async () => {
    sandbox = await createSandbox();
    output = await execPkgdrop(`init -y --offline`);
  });

  afterAll(async () => {
    sandbox.clean();
  });

  test('displays console messages', () => {
    expect(output).toMatchSnapshot();
  });

  test('files exist', async () => {
    expect(await sandbox.exists('pkgdrop.config.js')).toBe('file');
    expect(await sandbox.exists('-/')).toBe('dir');
    expect(await sandbox.exists('-/importmap.json')).toBe('file');
  });

  test('file content', async () => {
    expect(await sandbox.read('-/importmap.json')).toMatchSnapshot();
    expect(await sandbox.read('pkgdrop.config.js')).toMatchSnapshot();
  });
});
