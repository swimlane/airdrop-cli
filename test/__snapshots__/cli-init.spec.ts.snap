// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init displays console messages 1`] = `
"Writing pkgdrop.config.js
Writing importmap
Done! [XX s]"
`;

exports[`init file content 1`] = `
"{
  \\"imports\\": {},
  \\"scopes\\": {}
}"
`;

exports[`init file content 2`] = `
"module.exports = {
  \\"package_path\\": \\"./-/\\",
  \\"package_root\\": \\"/-/\\"
}"
`;
